alacritty_terminal/src/term/mod.rs

```diff
@@ -456,15 +465,33 @@ impl<'a, C> Iterator for RenderableCellsIter<'a, C> {
     #[inline]
     fn next(&mut self) -> Option<Self::Item> {
         loop {
-            if self.cursor_offset == self.inner.offset() && self.inner.column() == self.cursor.col {
-                let selected = self.is_selected(Point::new(self.cursor.line, self.cursor.col));
+            if self.cursor.point.line == self.inner.line()
+                && self.cursor.point.col == self.inner.column()
+            {
+                let selected = self.is_selected(self.cursor.point);
 
                 // Handle cursor
-                if let Some(cursor_key) = self.cursor_key.take() {
+                if self.cursor.rendered {
+                    let mut cell =
+                        RenderableCell::new(self.config, self.colors, self.inner.next()?, selected);
+
+                    if self.cursor.style == CursorStyle::Block {
+                        std::mem::swap(&mut cell.bg, &mut cell.fg);
+
+                        if let Some(color) = self.config.cursor_text_color() {
+                            cell.fg = color;
+                        }
+                    }
+
+                    return Some(cell);
+                } else {
+                    self.cursor.rendered = true;
```

Comment:
This `self.cursor.rendered` thing doesn't look nice tbh. I don't like old code here either. I have no suggestions yet though.

Reply:
I don't see what's wrong with it tbh. We need to somehow have a state between method calls to make sure the cursor and the cell are both rendered. A bool is probably the simplest way to store state.